### ISSUE TITLE
Bad error message while using kubectl expose service without specifying port

### ISSUE THREAD

**Issue by nikhiljindal** [GoogleCloudPlatform/kubernetes#11392]
I tried to create a service from another service using kubectl expose, without specifying the port and here is the error message I got:

```
error: it is not possible to get ports from
```
The command I ran:
```
./cluster/kubectl.sh expose service frontend --name=frontend2
```

The error is expected since I didnt specify a port (which is required), but the error message can be better.

**Comment by nikhiljindal** [comment]
cc @bgrant0607

**Comment by nikhiljindal** [comment]
FWIW, creating a service from rc works without specifying the port (even tough it is required as per https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/user-guide/kubectl/kubectl_expose.md)

Running:
```
kubectl expose rc frontend
```
works fine.

**Comment by nikhiljindal** [comment]
Discussed with @bgrant0607 We should fix the error message in 1.0.
Later, we should allow "kubectl expose myservice --name=newservice" (without specifying port), but that will be post 1.0
